packages/beacon-node/src/chain/stateCache/stateContextCache.ts

```diff
@@ -85,6 +85,29 @@ export class StateContextCache implements BlockStateCache {
     return this.cache.size;
   }
 
+  updateUnfinalizedPubkeys(validators: UnfinalizedPubkeyIndexMap): void {
+    let totalNumStatesUpdated = 0;
+
+    const addTimer = this.metrics?.addPubkeyTime.startTimer();
+    // Don't use a try/catch, only count run without exceptions
+    if (this.cache.size > 0) {
+      const st = this.cache.values().next().value as CachedBeaconStateAllForks;
+      // Only need to insert once since finalized cache is shared across all states globally
+      st.epochCtx.addFinalizedPubkeys(validators, this.metrics ?? undefined);
```

Comment:
since`epochCtx.addFinalizedPubkeys` should be called once, it should not be called here. Call it from the consumer, i.e `queued.ts` instead